pvlib/irradiance.py

```diff
@@ -31,6 +30,90 @@
                    'sea': 0.06}
 
 
+def cloud_opacity_factor(irr_dif_clouds: np.ndarray,
+                         irr_dir_clouds: np.ndarray,
+                         irr_ghi_clouds: np.ndarray,
+                         spectra: dict) -> (np.ndarray, np.ndarray):
+    """
+        Calculate the effect of "cloud opacity factor" on spectral
+        irradiance under clear sky.
+
+        First we calculate the rho fraction based on campbell_norman
+        irradiance with clouds converted to POA irradiance. In the
+        paper [1] these values are obtained from observations. The equations
+        used for calculating cloud opacity factor to scale the clear sky
+        spectral estimates using spectrl2. Results can be compared
+        with sun calculator:
+        https://www2.pvlighthouse.com.au/calculators/solar%20
+        spectrum%20calculator/solar%20spectrum%20calculator.aspx
+
+        Parameters
+        ----------
+        irr_dif_clouds:np.ndarray
+            Total diffuse irradiance (poa_diffuse) estimated using
+            `pvlib.irradiance.get_total_irradiance` and
+            `pvlib.irradiance.campbell_norman` irradiance with clouds
+            (transmittance)
+
+        irr_dir_clouds:np.ndarray
+            Total direct irradiance (poa_direct) estimated using
+            `pvlib.irradiance.get_total_irradiance` and
+            `pvlib.irradiance.campbell_norman` irradiance with
+            clouds (transmittance)
+
+        irr_ghi_clouds:np.ndarray
+            Total direct irradiance (poa_global) estimated
+            using `pvlib.irradiance.get_total_irradiance` and
+            `pvlib.irradiance.campbell_norman irradiance
+            with clouds (transmittance)
+
+        spectra:np.ndarray
+            Spectral irradiance output from `pvlib.spectrum.spectrl2`
+            under clear-sky conditions
```

Comment:
1. Similar to above, I'm not sure that this input needs a recommended source (note that [1] does not use spectrl2).  
2. I think the output of `spectrl2` is actually a dict, not an `ndarray`.  But would it make sense to take the spectral inputs as individual parameters (i.e. have the user pass in `spectra['poa_sky_diffuse'][:, 0]` etc instead of `spectra`)?  Although pvlib often returns many outputs bundled together for convenience, inputs are usually kept separate.